dolphin/db/sqlalchemy/api.py

```diff
@@ -114,10 +114,28 @@ def storage_create(storage):
     return storage_ref
 
 
+def registered_device_list(storage):
+    storage_ref = models.RegisteredDeviceList()
+    storage_ref.id = storage.id
+    storage_ref.name = storage.name
+    this_session = get_session()
+    this_session.begin()
+    this_session.add(storage_ref)
+    this_session.commit()
+    return storage_ref
+
+
 def storage_get(storage_id):
     this_session = get_session()
     this_session.begin()
     storage_by_id = this_session.query(Storage) \
         .filter(Storage.id == storage_id) \
         .first()
     return storage_by_id
+
+
+def get_registered_device_list():
```

Comment:
same here, use regstry_context get.. 
A sample for DB API client is below..
import sys

from dolphin import  db
from oslo_config import cfg

from dolphin.db.sqlalchemy.models import RegistryContext, Storage

CONF = cfg.CONF
storage_id='1234567322813900232034312213424'
register_info = RegistryContext(storage_id=storage_id, username='admin', password='password', hostname="host1", extra_attributes={'key1': 'value1', 'key2': 'value2"'})
storage = Storage(name='VMAX_Array1',id=storage_id,vendor='EMC',model='VMAX',description='EMC VMax Array')

def main():
    # Initialize DB Connection
    CONF(sys.argv[1:], project='dolphin',
         version='v1')
    db.register_db()
    # Create a Storage backend object in DB
    db.registry_context_create(register_info)
    db.storage_create(storage)
    # Get a Storage backend object by providing ID
    strorage=db.storage_get(storage_id)
    print(strorage)

    registry_context = db.registry_context_get(storage_id)
    print(registry_context)

    registry_contexts = db.registry_context_get_all()
    print()

if __name__ == '__main__':
    main()

Reply:
ok done.